### ISSUE TITLE
[SODA] Form field validation (header)

### ISSUE THREAD

**Issue by aaron-collier** [sul-dlss/hungry-hungry-hippo#909]
*Note:* I'm unclear if this is a duplicate or a different issue at this point:

The form field with an error is not identified in text, or text alternative, or via programmatic association.
There are several issues with the form validation.

- When a field is in error, an error message appears visibly inline, but it is not programmatically associated with a form field and disappears when a user begins interacting with the form. 
- When the error is a required field not entered, the text is a generic "Please fill out this field."
- Fields that are in error do not have aria-invalid="true" set on them.

There is also the issue that there are different types of error management which can be very confusing for users.

Soda recommendation:
```
The form field must be associated with the error text via aria-describedby as outline in the guidance that follows.

The error message text must be descriptive regarding the error and the form field in error. 

When a form field is error, aria-invalid="true" should be added.

Ensure there is one type of error management across the site.

WebAIM - Usable and Accessible Form Validation and Error Recovery
https://webaim.org/techniques/formvalidation/

RULE : 
Text-based alerts that identify the field in error MUST be provided to users for form validation cues and errors.

HOW TO FIX: 
Fix this issue by doing AT LEAST ONE of the following:
1. Programmatically associate the error message and the field in error - typically using the aria-describedby attribute to reference a unique id attribute value on the error message container. (STRONGLY PREFERRED)
2. Include the field name in the error message.

REFERENCE: 
Deque University: https://dequeuniversity.com/class/forms/form-validation/error-identification
W3C-WAI tutorial: https://www.w3.org/WAI/tutorials/forms/notifications/

BACKGROUND: 
When form fields that have an error are not identified with a text-based error message people who are blind or have low vision may not be able to understand that an error has occurred and what went wrong. In order to provide feedback that can be accessed by all users, error messages must be provided in a text format. Error messages must indicate which input is in error - either by the text of the error message or programmatically - and the nature of the error. This will allow people to recover from mistakes while filling out a form.

```

See issue id 2228 in https://docs.google.com/spreadsheets/d/1x0uBBDe3T8O0HNCs74OdVKQc8tQQNvOV_f-C6IS44IQ/edit?gid=1050712151#gid=1050712151

**Comment by thatbudakguy** [comment]
Just noting so I remember here that we determined this issue applies _only_ to the contact form accessible from the header, not to the work/collection forms.

**Comment by thatbudakguy** [comment]
We determined that we're not going to fix/alter the behavior of the error messages here because they're browser default behavior, so we're not going to try to layer something custom on top of it. #1121 fixes the rest of this issue.
